### ISSUE TITLE
pause and resume methods

### ISSUE THREAD

**Issue by dp-thomas** [dixonandmoe/rellax#58]
Maybe you could add `pause` and `resume` method. This could already do the trick:

        self.resume = function () {
            pause = false;  
        };

        self.pause = function () {
            pause = true;
        };

**Comment by c0mrx** [comment]
I agree!

Right now I destroy the rellax element on smaller viewports and it'd be great to have a vanilla way of recreating the rellax when the browser viewport sizes back up.

I've tried a few things, but binding `  var rellax = new Rellax('.rellax', {
    center: true
  });` to a `$(window).resize` causes it to keep creating instances every time the viewport is resized, pushing my elements up and down the page more and more as you resize.

**Comment by Glumanda99** [comment]
@c0mrx May I know why you destroy rellax on smaller viewports?

**Comment by c0mrx** [comment]
@Glumanda99 The elements were moving too much and covering more important information, so for UX reasons I've disabled it on smaller screens.

**Comment by Glumanda99** [comment]
@c0mrx Thanks for your fast response! I'm not sure if I'm gonna have the same issues while designing, but I could imagine to remove the `rellax` class on smaller viewports and add them back on all `data-rellax` attributes when appropriate. Need to try this..

**Comment by Glumanda99** [comment]
I think I also need to reset `translate3d` every time I do this.

**Comment by c0mrx** [comment]
@Glumanda99 I ended up manipulating the speed attribute. It's not the cleanest way to do it, but it does work.

```
  // Init Rellax.js after all elements have fully loaded so that all heights of DOMs are calculated
  window.onload = function() {
    var rellax = new Rellax('.rellax', {
      center: true
    });
    // 'Disable' rellax on mobile
    function checkWidth() {
      var windowSize = $(window).innerWidth();
      if (windowSize <= 960) {
        $('.rellax').attr('data-rellax-speed', '0');
        rellax.refresh();
      } else {
        rellax.refresh();
        $('.even.rellax').attr('data-rellax-speed', '-1');
        $('.odd.rellax').attr('data-rellax-speed', '1');
      }
    }
    // Execute on load
    checkWidth();
    // Refresh whenever window is resized.
    $(window).resize(function() {
      checkWidth();
    });
  };
```

**Comment by captainfromspace** [comment]
@c0mrx, in your code example above you are calling `rellax.refresh()`, is that a custom method you've written? I need to refresh the rellax after an AJAX-call that adds content, but destroying and re-initializing it is not working that good[].

[1] - https://github.com/dixonandmoe/rellax/issues/91

**Comment by sergeymorkovkin** [comment]
I'm trying to use Rellax with the [OnScreen](https://github.com/silvestreh/onScreen). This is mainly required to achieve acceptable page performance. Therefore, I want to pause/suspend all rellax instances except for those that are visible in the viewport.

_Unfortunately, I couldn't achieve this without copying rellax src code and patching it._

My recommendation is implementing two methods: attach() and detach(). All it should do is:

1. Stop requestAnimationFrame loop;
2. Detach window.onresize handler;

All this should be done WITHOUT reverting initial element styles. Otherwise it'll all jump-flicker.

**Comment by zachtownsend** [comment]
Here's what I've used to make it run responsively:
```
const getRellaxSpeed = () => $window.width() < 768 ? 0 : -7;
const rellax = new Rellax('.rellax', {
    speed: getRellaxSpeed(),
});

let throttle = null;

$window.resize(() => {
    clearTimeout(throttle);

    throttle = setTimeout(() => {
        rellax.options.speed = getRellaxSpeed();
        rellax.refresh();
    }, 400);
});
```
(The throttle thing is not necessary, but you get the idea)

**Comment by sergeymorkovkin** [comment]
> Here's what I've used to make it run responsively:

Yeah, that looks good for your goals. In my situation, however, I need to reduce the use of animation frames. Otherwise it makes smooth scrolling jerky. So, I still need detach/attach. Preferrably, persisting Rellax instance.
